src/run/main.c

```diff
@@ -1,19 +1,68 @@
-//-----------------------------------------------------------------------------
-//   main.c
-//
-//   Project:  EPA SWMM5
-//   Version:  5.1
-//   Date:     05/10/2018
-//   Author:   L. Rossman
-
-//   Main stub for the command line version of EPA SWMM 5.1
-//   to be run with swmm5.dll.
+/*
+ *  main.c - Main stub for the command line version of EPA SWMM 5.1
+ *
+ *  Created on: October 9, 2020
+ *  Updated on:
+ *
+ *  Author:     Michael E. Tryby
+ *              US EPA - ORD/CESER
+ *
+ */
 
+// System includes
 #include <stdio.h>
 #include <string.h>
-#include <time.h>
+#include <math.h>
 
+// Private project includes
+#include "timer.h"
+
+// Public project includes
 #include "swmm5.h"
+#include "toolkitAPI.h"
+
+
+#define BAR_LEN 50l
+#define MSG_LEN 84
+
+static long Start;
+
+
+void write_console(char *msg)
+{
+    fprintf(stdout, "%s", msg);
+    fflush(stdout);
+}
+
+void progress_bar(double *ratio)
+{
+    char bar[BAR_LEN + 1] = {'\0'};
+
+    // Create progress bar
+    long prog_len = lround(*ratio * BAR_LEN);
+    memset(bar, ' ', (size_t)BAR_LEN);
+    if (prog_len < BAR_LEN)
+        memset(bar, '>', (size_t)(prog_len + 1));
+    memset(bar, '=', (size_t)prog_len);
+
+    // Compute percent complete
+    double pct = *ratio * 100.0;
+
+    // Compute time remaining using naive approach
+    long t_r = 0;
+    if (*ratio > 0.0)
+        t_r = lround((1.0 - *ratio) * (current_time_millis() - Start) / *ratio);
```

Comment:
@michaeltryby, Is there a reason you are using time instead of percent of the simulation period remaining?

Reply:
@cbuahin The code highlighted is a naive estimate of the wall time remaining until the simulation is complete. This gives the user an indication of when the simulation will finish when they are running long simulations. It uses a simple ratio based approach. Cumulative wall time and progress are used to dampen out minor fluctuations in simulation speed/progress. It seems to work alright for short simulations and how not been tested for longer ones.